### PR TITLE
Overmap Radio Beacons

### PR DESCRIPTION
**Pull request by comma**
Not actually used anywhere yet but all is working.
Basically just a small overmap object that can attach to another object and move with it, and give a message when scanned via sensors consoles etc.
Added an item for spawning those, but item itself is not accessible ingame.
Could be used for SOS and crap but lazy.
image

### CODE REVIEW

File: code/modules/overmap/radio_beacon.dm [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	icon = 'icons/obj/radio.dmi'
+	icon_state = "walkietalkie"
+	var/obj/effect/overmap/radio/signal
+
+/obj/item/radio_beacon/attack_self(mob/user)
+	var/obj/effect/overmap/visitable/O = map_sectors["[get_z(src)]"]
+	if(!O)
+		to_chat(user, SPAN_WARNING("You cannot deploy [src] here."))
+		return
+	var/message = sanitize(input("What should it broadcast") as message|null)
```
Comment (MistakeNot4892) [inline code/modules/overmap/radio_beacon.dm]: Needs a question mark.

Comment (comma) [reply code/modules/overmap/radio_beacon.dm]: Added

**Review by MistakeNot4892 — APPROVED** [review]

**Comment by MistakeNot4892** [comment]
Only thing I can think of is that you can't turn the beacon off again afterwards.

**Comment by comma** [comment]
It's more or less a debug item anyway, can just del it to clean up the signal

File: code/modules/overmap/radio_beacon.dm [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	if(!O)
+		to_chat(user, SPAN_WARNING("You cannot deploy [src] here."))
+		return
+	var/message = sanitize(input("What should it broadcast?") as message|null)
+	
+	if(!signal)
+		signal = new()
+	
+	signal.message = message
+	signal.set_origin(O)
```
Comment (afterthought2) [inline code/modules/overmap/radio_beacon.dm:50]: I don't get it; what if I take this obj once deployed and move it to a z level with a different overmap thing associated with it?

Comment (comma) [reply code/modules/overmap/radio_beacon.dm:50]: Then it doesn't follow. It's just a debug item, it's not very robust

Comment (MistakeNot4892) [reply code/modules/overmap/radio_beacon.dm:50]: Can be removed if desired, it's not on the map or craftable so doesn't seem like an issue.

**Pull request merged**